cryptotest/tests/MacTests.java

```diff
@@ -67,23 +67,39 @@ protected void checkAlgorithm(Provider.Service service, String alias) throws Alg
         try {
             Mac md = Mac.getInstance(alias, service.getProvider());
             byte[] b = new byte[]{1, 2, 3};
+            Provider provider =  service.getProvider();
+            String algorithm = service.getAlgorithm();
+            String generatorAlgorithm;
 
-            if (service.getAlgorithm().contains("PBE")) {
+            if (algorithm.contains("PBE")) {
```

Comment:
Why this?

Reply:
I don't understand the question. I introduced some variables, so I would not be copying the same expression over multiple places.

Reply:
I do not understand the question too.

Reply:
:)